runtime/src/iree/hal/cts/device_group_copy_test.h

```diff
@@ -0,0 +1,721 @@
+}
+
+TEST_F(DeviceGroupCopyTest, CopyBetweenDevicesFromQueue1) {
+  iree_hal_command_buffer_t* command_buffer1 = NULL;
+  IREE_ASSERT_OK(iree_hal_command_buffer_create(
+      device_, IREE_HAL_COMMAND_BUFFER_MODE_ONE_SHOT,
+      IREE_HAL_COMMAND_CATEGORY_TRANSFER, 0x2,
+      /*binding_capacity=*/0, &command_buffer1));
+
+  iree_hal_command_buffer_t* command_buffer2 = NULL;
+  IREE_ASSERT_OK(iree_hal_command_buffer_create(
+      device_, IREE_HAL_COMMAND_BUFFER_MODE_ONE_SHOT,
+      IREE_HAL_COMMAND_CATEGORY_TRANSFER, 0x2,
+      /*binding_capacity=*/0, &command_buffer2));
+  uint8_t i8_val = 0x54;
+  std::vector<uint8_t> reference_buffer(kDefaultAllocationSize);
+  std::memset(reference_buffer.data(), i8_val, kDefaultAllocationSize);
+
+  // Create and fill a host buffer.
+  iree_hal_buffer_params_t host_params = {0};
+  host_params.type =
+      IREE_HAL_MEMORY_TYPE_HOST_LOCAL | IREE_HAL_MEMORY_TYPE_DEVICE_VISIBLE;
+  host_params.usage = IREE_HAL_BUFFER_USAGE_DISPATCH_STORAGE |
+                      IREE_HAL_BUFFER_USAGE_TRANSFER |
+                      IREE_HAL_BUFFER_USAGE_MAPPING;
+  host_params.queue_affinity = 0x2;
+  iree_hal_buffer_t* host_buffer = nullptr;
+  IREE_ASSERT_OK(iree_hal_allocator_allocate_buffer(
+      device_allocator_, host_params, kDefaultAllocationSize, &host_buffer));
+  IREE_ASSERT_OK(iree_hal_device_transfer_h2d(
+      device_, reference_buffer.data(), host_buffer, 0, reference_buffer.size(),
+      IREE_HAL_TRANSFER_BUFFER_FLAG_DEFAULT, iree_infinite_timeout()));
+
+  // Create a device buffer.
+  iree_hal_buffer_params_t device_params = {0};
+  device_params.type =
+      IREE_HAL_MEMORY_TYPE_DEVICE_LOCAL | IREE_HAL_MEMORY_TYPE_HOST_VISIBLE;
+  device_params.usage = IREE_HAL_BUFFER_USAGE_DISPATCH_STORAGE |
+                        IREE_HAL_BUFFER_USAGE_TRANSFER |
+                        IREE_HAL_BUFFER_USAGE_MAPPING;
+  device_params.queue_affinity = 0x2;
+
+  iree_hal_buffer_t* device_buffer_device_1 = nullptr;
+  IREE_ASSERT_OK(iree_hal_allocator_allocate_buffer(
+      device_allocator_, device_params, kDefaultAllocationSize,
+      &device_buffer_device_1));
+
+  device_params.queue_affinity = 0x2;
+  iree_hal_buffer_t* device_buffer_device_2 = nullptr;
+  IREE_ASSERT_OK(iree_hal_allocator_allocate_buffer(
+      device_allocator_, device_params, kDefaultAllocationSize,
+      &device_buffer_device_2));
+
+  // Copy the host buffer to the device buffer.
+  IREE_ASSERT_OK(iree_hal_command_buffer_begin(command_buffer1));
+  IREE_ASSERT_OK(iree_hal_command_buffer_copy_buffer(
+      command_buffer1, /*source_ref=*/
+      iree_hal_make_buffer_ref(host_buffer, 0, kDefaultAllocationSize),
+      /*target_ref=*/
+      iree_hal_make_buffer_ref(device_buffer_device_1, 0,
+                               kDefaultAllocationSize)));
+  IREE_ASSERT_OK(iree_hal_command_buffer_end(command_buffer1));
+
+  IREE_ASSERT_OK(SubmitCommandBufferAndWait(
+      command_buffer1, iree_hal_buffer_binding_table_empty(), 0x2));
```

Comment:
do you mean 0x1?
(it'd be better to hoist these up into consts and use them as kQueue0 and kQueue1)